VERSION-RANGE-SPEC.rst

```diff
@@ -0,0 +1,989 @@
+  such as sat solvers) that are ecosystem- and tool-specific and outside of the
+  scope for this specification; yet ``vers`` could be used in tandem with
+  ``purl`` to provide an input to this dependencies resolution process.
+
+
+Examples
+~~~~~~~~~
+
+A single version in an npm package dependency:
+
+- originally seen as a dependency on version "1.2.3" in a package.json manifest
+- the version range spec is: ``vers:npm/1.2.3``
+
+
+A list of versions, enumerated:
+
+- ``vers:pypi/0.0.0|0.0.1|0.0.2|0.0.3|1.0|2.0pre1``
+
+
+A complex statement about a vulnerability in a "maven" package that affects
+multiple branches each with their own fixed versions at 
+https://repo1.maven.org/maven2/org/apache/tomee/apache-tomee/ 
+Note how the constraints are sorted:
+
+
+- "affects Apache TomEE 8.0.0-M1 - 8.0.1, Apache TomEE 7.1.0 - 7.1.2,
+  Apache TomEE 7.0.0-M1 - 7.0.7, Apache TomEE 1.0.0 - 1.7.5."
+
+- a normalized version range spec is:
+  ``vers:tomee/>=1.0.0-beta1|<=1.7.5|>=7.0.0-M1|<=7.0.7|>=7.1.0|<=7.1.2|>=8.0.0-M1|<=8.0.1``
+
+- alternatively, four ``vers`` express the same range, using one ``vers`` for
+  each vulnerable "branches": 
+  - ``vers:tomee/>=1.0.0-beta1|<=1.7.5``
+  - ``vers:tomee/>=7.0.0-M1|<=7.0.7``
+  - ``vers:tomee/>=7.1.0|<=7.1.2``
+  - ``vers:tomee/>=8.0.0-M1|<=8.0.1``
+
+Conversing Rubygems custom syntax for dependency on gem. Note how the
+pessimistic version constraint is expanded:
+
+- ``'library', '~> 2.2.0', '!= 2.2.1'``
+- the version range spec is: ``vers:gem/>=2.2.0|!= 2.2.1|<2.3.0``
+
+
+URI scheme
+~~~~~~~~~~
+
+The ``vers`` URI scheme is  an acronym for "VErsion Range Specifier".
+It has been selected because it is short, obviously about version and available
+for a future formal registration for this URI-scheme at the IANA registry.
+
+The URI scheme is followed by a colon ":".
+
+
+``<versioning-scheme>``
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``<versioning-scheme>`` (such as ``npm``, ``deb``, etc.) determines:
+
+- the specific notation and conventions used for a version string encoded in
+  this scheme. Versioning schemes often specify a version segments separator and
+  the meaning of each version segments, such as [major.minor.patch] in semver.
+
+- how two versions are compared as greater or lesser to determine if a version
+  is within or outside a range.
+
+- how a versioning scheme-specific range notation can be transformed in the
+  ``vers`` simplified notation defined here.
+
+By convention the versioning scheme **should** be the same as the ``Package URL``
+package type for a given package ecosystem. It is OK to have other schemes
+beyond the purl type. A scheme could be specific to a single package name.
+
+The ``<versioning-scheme>`` is followed by a slash "/".
+
+
+``<version-constraint>``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+After the ``<versioning-scheme>`` and "/" there are one or more
+``<version-constraint>`` separated by a pipe "|". The pipe "|" has no special
+meaning beside being a separator.
```

Comment:
I would suggest to prohibit 2 consecutive pipes - as this would generate an empty `<version-contraint>`. That makes it much easier to parse and check. There more specific the spec is, the easier it is to implement.
```suggestion
meaning beside being a separator. An empty ``<version-contraint>`` is invalid. Therefore, two consecutive pipes "|" SHALL not exist.